src/lib.rs

```diff
@@ -467,21 +467,31 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
 
                     let mut ui = ui.child_ui(rect, Default::default());
                     ui.push_id(node_index, |ui| {
```

Comment:
@emilk How about basing the ID on `tab` rather than `node_index`? This would require some kind of API change since not all tab types can be used as an `Id`. Perhaps adding a new method to `TabViewer` like `fn tab_id(&self, tab: &Self::Tab) -> egui::Id;`?

Reply:
Sure, but then I would suggest returning a default value for it (the same for everything). That would in effect put all tabs into the same "`Id` space" for users who don't want or care about implementing `tab_id`.

But philosophically I feel this is feature creep. As a user, I want `egui_dock` to give me docking, and do it well, and nothing more. I can do my own `push_id` call if I want to, without having to pass `Id's` to `egui_dock`. Same with `ScrollArea`: I can do it myself, and have access to all the options of `ScrollArea` etc. But - I can see the point in a more "batteries included" approach libraries too, it just tends to keep creeping (e.g. should `egui_dock` implement custom frames for each tab? Perhaps custom `Style`:s for each tab? etc)